src/frame/js/actions/auth.js

```diff
@@ -1,33 +1,85 @@
+import { batchActions } from 'redux-batched-actions';
+
 import http from './http';
-import { handleUserConversationResponse } from './conversation';
+import { resetConversation, handleUserConversationResponse, disconnectFaye } from './conversation';
+import { setUser, resetUser } from './user';
+import { resetIntegrations } from './integrations';
 
-import { getClientInfo } from '../utils/client';
+import { getClientId, getClientInfo } from '../utils/client';
 import { removeItem } from '../utils/storage';
 
 export const SET_AUTH = 'SET_AUTH';
 export const RESET_AUTH = 'RESET_AUTH';
 
-export function login() {
+export function login(userId, jwt) {
     return (dispatch, getState) => {
-        const {config: {appId}, user: {userId}, auth: {sessionToken}} = getState();
+        const {config: {appId}, auth: {sessionToken}} = getState();
+
+        const actions = [
+            setAuth({
+                jwt
+            }),
+            setUser({
+                userId
+            }),
+            resetConversation(),
+            resetIntegrations()
+        ];
+
+        dispatch(disconnectFaye());
+        dispatch(batchActions(actions));
+
         return dispatch(http('POST', `/apps/${appId}/login`, {
             userId,
             sessionToken,
             client: getClientInfo(appId)
-        })).then(({response, ...props}) => {
+        })).then((response) => {
             // get rid of session token
             removeItem(`${appId}.sessionToken`);
-            dispatch(setAuth({
-                sessionToken: null
-            }));
 
-            if (response.status === 200) {
-                return dispatch(handleUserConversationResponse(props));
+            if (response && Object.keys(response).length > 0) {
+                return dispatch(handleUserConversationResponse(response));
             }
         });
     };
 }
 
+export function logout() {
+    return (dispatch, getState) => {
+        const {config: {appId}, user: {_id: appUserId, userId}} = getState();
+
+        if (!userId) {
+            return Promise.resolve();
+        }
+
+        let promise;
+        if (appUserId) {
```

Comment:
It's because the SDK was initialized with a `userId`, but the user doesn't exist on the server yet. So yeah I guess that implies they haven't sent a message yet.

But it's also to have never sent a message and still have an `appUserId`, if the user was pre-created for example